media_driver/linux/common/ddi/media_libva_util.cpp

```diff
@@ -874,34 +876,36 @@ void* DdiMediaUtil_LockSurface(DDI_MEDIA_SURFACE  *surface, uint32_t flag)
             {
                 mos_bo_map(surface->bo, flag & MOS_LOCKFLAG_WRITEONLY);
             }
-            else if (surface->pMediaCtx->m_useSwSwizzling)
+            else if ((surface->pMediaCtx->m_useSwSwizzling) && !(flag & MOS_LOCKFLAG_NO_SWIZZLE))
```

Comment:
MOS_LOCKFLAG_NO_SWIZZLE flag is meaningful only when m_useSwizzling is true, right? So I think we should move the conditional statements about MOS_LOCKFLAG_NO_SWIZZLE into 'else if ((surface->pMediaCtx->m_useSwSwizzling)'.

Reply:
> OS_LOCKFLAG_NO_SWIZZLE flag is meaningful only when m_useSwizzling is true, right? 
No, please see line #901